pkg/daemon/daemon.go

```diff
@@ -950,6 +989,17 @@ func (p *ptpProcess) processPTPMetrics(output string) {
 				state = event.PTP_HOLDOVER // consider s1 state as holdover,this passed to event to create metrics and events
 			}
 			p.ProcessTs2PhcEvents(ptpOffset, source, ifaceName, state, values)
+		} else if clockState == HOLDOVER || clockState == LOCKED {
+			// in case of holdover without iface, still need to update clock class for T_G
+			if p.name != ts2phcProcessName && p.name != syncEProcessName { // TGM announce clock class via events
+				p.SetPmcCheck(false) // reset pmc check since we are updating clock class here
```

Comment:
Might be best to use consume for all times of setting it to false to make it easier to search of when it happens

Reply:
made changes as suggested